src/python/grpcio/grpc/_channel.py

```diff
@@ -981,4 +981,7 @@ def __del__(self):
         # then deletion of this grpc._channel.Channel instance can be made to
         # effect closure of the underlying cygrpc.Channel instance.
         cygrpc.fork_unregister_channel(self)
-        _moot(self._connectivity_state)
+        # This prevent the failed-at-initializing object removal from failing.
+        # Though the __init__ failed, the removal will still trigger __del__.
+        if hasattr(self, "_connectivity_state"):
```

Comment:
Is it possible to get this working by reordering statements in `__init__` or preinitializing with `None`?

Reply:
That's a better idea!

Reply:
The `self._connectivity_state` depends on `self._channel`. Since other classes in gRPC-Python is not doing the preinitializing with `None`, I would think a check in `__del__` is a little bit more consistent.
```Python
_channel.py@910
self._channel = cygrpc.Channel(_common.encode(target), _options(options), credentials)
self._call_state = _ChannelCallState(self._channel)
self._connectivity_state = _ChannelConnectivityState(self._channel)
```

Reply:
Does `_moot` work if `None` is passed? If yes, I think pre-initializing with `None` one and re-assigning it later is preferred. If `_moot` breaks on `None`, then I am okay with the current solution.

Reply:
`_moot` doesn't work for `None`:
`_moot` will access the passing variable's attribute `lock` which will cause another Exception.
```Python
Exception AttributeError: "'NoneType' object has no attribute 'lock'" in <bound method Channel.__del__ of <_open_source_grpc._channel.Channel object at 0x7f9b3d86c3d0>> ignore
```